src/Workspaces/Core/Portable/Workspace/Workspace_Editor.cs

```diff
@@ -571,9 +598,27 @@ protected internal void OnAnalyzerConfigDocumentOpened(DocumentId documentId, So
             this.RegisterText(textContainer);
         }
 
+        /// <summary>
+        /// Tries to close the document identified by <paramref name="documentId"/>.  This is only needed by
+        /// implementations of ILspWorkspace to indicate that the workspace should try to transition to the closed state
+        /// for this document, but can bail out gracefully if they don't know about it (for example if they haven't
+        /// heard about the file from the project system).  Subclasses should determine what file contents they should
+        /// transition to if the file is within the workspace.
+        /// </summary>
+        /// <param name="documentId"></param>
+        internal virtual void TryOnDocumentClosed(DocumentId documentId)
+        {
+            throw new NotImplementedException();
+        }
```

Comment:
this is only called for an ILspWorkspace, and only in the 'mutating' form (currently only the TestWorkspace, and potentially any other mutating workspaces we might hypothetically add in the future).  So it's fine to just throw here.